packages/context/src/context.ts

```diff
@@ -11,8 +11,9 @@ export type Constructor<T> = new(...args) => T;
 export class Context {
   private registry: Map<string, Binding>;
 
-  constructor() {
+  constructor(private _parent?: Context) {
     this.registry = new Map();
```

Comment:
Interesting idea. I think I could use that unique id in my test instead of `@inject('ctx')`.

@raymondfeng How do you propose to generate that unique id? Should we just increase a number, or should we use `uuid`?

Do you have any other comments, or does the rest of the patch look good to you?